cloudplatform/connectivity-apache-httpclient5/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultApacheHttpClient5Factory.java

```diff
@@ -233,4 +238,42 @@ private boolean isValidProxyConfigurationUriInDestination( @Nullable final HttpD
         }
         return true;
     }
+
+    private static class LoggingHttpRequestRetryStrategy extends DefaultHttpRequestRetryStrategy
+    {
+        final @Nullable HttpDestinationProperties destination;
+
+        public LoggingHttpRequestRetryStrategy( final @Nullable HttpDestinationProperties destination )
+        {
+            super();
+            this.destination = destination;
+        }
+
+        @Override
+        public boolean retryRequest( final HttpResponse response, final int execCount, final HttpContext context )
+        {
+            final boolean retry = super.retryRequest(response, execCount, context);
+            if( retry ) {
+                final String msg = "Retrying request for destination {} due to response {}. Retry attempt #{}.";
+                log.debug(msg, destination, response.getCode(), execCount);
+            }
+            return retry;
+        }
+
+        @Override
+        public boolean retryRequest(
+            final HttpRequest req,
+            final IOException exception,
+            final int execCount,
+            final HttpContext context )
+        {
+            final boolean retry = super.retryRequest(req, exception, execCount, context);
+            if( retry ) {
+                final String msg =
+                    "Retrying {} request for destination {} to {} due to exception \"{}\". Retry attempt #{}.";
+                log.debug(msg, req.getMethod(), destination, req.getRequestUri(), exception.getMessage(), execCount);
```

Comment:
should be at least warn, maybe even error
```suggestion
                log.warn(msg, req.getMethod(), destination, req.getRequestUri(), exception.getMessage(), execCount);
```